news-explorer/src/components/PopupWithForm/PopupWithForm.js

```diff
@@ -1,57 +1,113 @@
 import './PopupWithForm.css';
-function PopupWithForm({isSignUpOpen, onSignUpClick}) {
+import {useForm} from '../../formHooks/useForm';
+import {useEffect, useRef} from 'react';
+function PopupWithForm({
+	isSignUpOpen,
+	isSignInOpen,
+	onSubmit,
+	onSignupSubmit,
+	setUsername,
+	setUserEmail,
+	setUserPassword,
+	errorMessage,
+}) {
+	const {handleInputChange, values, isValid, errors, resetForm} = useForm();
+
+	function handleErrorVisible() {
+		if (!isValid) {
+			return 'popup__input-error popup__input-error_visible';
+		} else {
+			return 'popup__input-error';
+		}
+	}
+
+	const button = useRef();
+
+	useRef(() => {
+		if (!isValid) {
+			button.current.setAttribute('disables', '');
+		} else {
+			button.current.removeAttribute('disables', '');
+		}
+	});
+
+	useEffect(() => {
+		if (!isSignInOpen || !isSignUpOpen) resetForm();
+	}, [isSignInOpen, isSignUpOpen]);
+
 	return (
 		<>
```

Comment:
`<>` empty tags are not needed if you have a common tag that wraps the whole markup of the component